File: src/main/java/com/vaadin/demo/component/dashboard/DashboardWidgetContents.java
package com.vaadin.demo.component.dashboard;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.dashboard.DashboardWidget;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.router.Route;

@Route("dashboard-widget-contents")
public class DashboardWidgetContents extends Div {

    public DashboardWidgetContents() {
        // tag::snippet[]
        DashboardWidget widget = new DashboardWidget("Widget title");
        widget.setContent(new Span("Widget content"));
        widget.setHeaderContent(new Span("Additional header content"));
        // end::snippet[]

        add(widget);
    }

    public static class Exporter extends DemoExporter<DashboardWidgetContents> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/dashboard/dashboard-widget-contents.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/dashboard/vaadin-dashboard-widget.js';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('dashboard-widget-contents')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-dashboard-widget widget-title="Widget title">
        <span>Widget content</span>
        <span slot="header-content">Additional header content</span>
      </vaadin-dashboard-widget>
      <!-- end::snippet[] -->
    `;
  }
}
